modules/tinymce/src/core/main/ts/api/Mode.ts

```diff
@@ -1,18 +1,33 @@
 import { Cell, Fun } from '@ephox/katamari';
 
-import { registerMode, setMode } from '../mode/Mode';
+import { getEditorReadOnlyProperty, registerMode, setMode } from '../mode/Mode';
 import { isReadOnly, registerReadOnlyContentFilters, registerReadOnlySelectionBlockers } from '../mode/Readonly';
 import Editor from './Editor';
 
+export interface ReadOnlyProperty {
+  selectionEnabled?: boolean;
+}
+
+export type EditorReadOnlyType = boolean | ReadOnlyProperty;
+
 /**
  * TinyMCE Editor Mode API.
  *
  * @class tinymce.EditorMode
  */
 
 export interface EditorMode {
+
   /**
-   * Checks if the editor is in a readonly state.
+   * Checks if the editor content can be selected.
+   *
+   * @method isSelectionEnabled
+   * @return {boolean} true if the editor content area allows selection.
+   */
+  isSelectionEnabled: () => boolean;
```

Comment:
This is a very late thought, I apologise, but you have _significantly_ improved the stability of selection readonly with this branch.

I wonder if we should have confidence in our change and just make this this always true rather than needing a custom readonly mode to activate selection.

It would certainly simplify the comments changes. I've posted a message in Slack to get some product input.